packages/react-lowcode/src/codegen/generation/generators/list/mui-datatable-generator.ts

```diff
@@ -98,6 +99,80 @@ export default class MuiDataTableGenerator implements TableGenerator
       return alteredSource
     }
 
+    async getColumnSourcePosition(position: SourceLineCol,
+                                  columnIndex: number): Promise<ColumnSourcePositionResult | undefined> {
+      let result: ColumnSourcePositionResult | undefined;
+
+      if (this._widgetContext) {
+        let sourceCode = await this._widgetContext.getSourceCodeString(position)
+
+        let ast = createAst(sourceCode)
+
+        if (ast) {
+          let widgetParentNode = findWidgetParentNode(sourceCode, position)
+
+          if (widgetParentNode) {
+            let columnsDeclarationNode = this.findColumnsDeclaration(widgetParentNode)
+
+            if (columnsDeclarationNode) {
+              let columnDeclarationArray = columnsDeclarationNode.getChildAt(2) as ts.ArrayLiteralExpression
```

Comment:
`getChiledAt(2)` seems to be hard to maintain (even buggy)
if it is necessary at least explain it in the comment

Reply:
yes it is ugly and there are lot of places which use similar code..i think after we would be able to generate whole app we have to refactor whole codegen